pkg/controllers/servicesync/servicesync.go

```diff
@@ -0,0 +1,198 @@
+package servicesync
+
+import (
+	"context"
+	"github.com/loft-sh/vcluster/pkg/constants"
+	"github.com/loft-sh/vcluster/pkg/controllers/resources/services"
+	"github.com/loft-sh/vcluster/pkg/util/loghelper"
+	corev1 "k8s.io/api/core/v1"
+	apiequality "k8s.io/apimachinery/pkg/api/equality"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+	"strings"
+)
+
+type ServiceSyncer struct {
+	SyncServices    map[string]types.NamespacedName
+	CreateNamespace bool
+
+	From ctrl.Manager
+	To   ctrl.Manager
+
+	Log loghelper.Logger
+}
+
+func (e *ServiceSyncer) Register() error {
+	reverseMapping := map[string]types.NamespacedName{}
+	for k, v := range e.SyncServices {
+		splitted := strings.Split(k, "/")
+		reverseMapping[v.Namespace+"/"+v.Name] = types.NamespacedName{
+			Namespace: splitted[0],
+			Name:      splitted[1],
+		}
+	}
+
+	return ctrl.NewControllerManagedBy(e.From).
+		Named("servicesync").
+		For(&corev1.Service{}).
+		Watches(source.NewKindWithCache(&corev1.Service{}, e.To.GetCache()), &serviceHandler{Mapping: reverseMapping}).
+		Complete(e)
+}
+
+func (e *ServiceSyncer) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	from := req.Namespace + "/" + req.Name
+	to, ok := e.SyncServices[from]
+	if !ok {
+		return ctrl.Result{}, nil
+	}
```

Comment:
Minor improvement suggestion: this could be implemented with [an event filter](https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.11.2/pkg/builder#Builder.WithEventFilter) based on [a predicate function](https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.11.2/pkg/predicate#NewPredicateFuncs), but admittedly the performance impact is negligible.